osu.Game/Screens/Edit/Compose/Components/BlueprintContainer.cs

```diff
@@ -113,8 +113,11 @@ protected override bool OnMouseDown(MouseDownEvent e)
             // check if selection has occurred
             if (selectionPerformed)
             {
-                // propagate right click to show context menu on selection
-                return e.Button != MouseButton.Right;
+                // only unmodified right click should show context menu
+                var shouldShowContextMenu = e.Button == MouseButton.Right && !e.ShiftPressed && !e.AltPressed && !e.SuperPressed;
```

Comment:
I'm not sure where I was coming from with my previous comment, but unless `ContextMenuContainer` was previewing input earlier than its normal handling flow, it wouldn't really help here.

In addition, the local handling in the editor *is* still required as an intercept, as something it wants to do things and rightfully block the context menu from displaying.

@bdach not sure I like your proposal any better than how we have things done right now, to be honest.

I'm going to take a look into this, but the solution in this PR may actually be okay enough.

Reply:
The hope was that even though the handling in this one place would still be awkward with my proposal, the other places that need to ignore right clicks so that it propagates correctly would go away. But now that I think about it they wouldn't, because my suggestion would only work if the right click initiates selection and would break right click on already selected objects. So I guess I'm indifferent at this point.